### DISCUSSION TITLE
Ethernet interface type equivalents in Netbox

### DISCUSSION THREAD

**Discussion by jayGG83** [netbox-community/netbox#13696] [Q&A]
Hello,

I'd like to have your insight to check if my ethernet interface types mapping are correct. I'm trying to create microwave devices in Netbox (my knowledge is a bit limited as I'm only a developer, not a telco engineer) and some of the interface type names not really match with the interface type names in the devices and I'd like to verify if the types I selected are the correct ones. In case not, which would be the best option?

| Interface type  | Netbox equivalent |
| ------------- | ------------- |
| 10/100 BASE-T  | 100BASE-T1  |
| 10/100/1000BASE-T  | 1000BASE-T (1GE)  |
| 100/1G BASE-T  | 1000BASE-T (1GE)  |
| 1000BASE-T/X  | 1000BASE-T (1GE)  |
| 1000/2500 BASE-T/X  | 2.5GBASE-T (2.5GE)  |
| 1G BASE-T/X  | SFP  |
| 1G/10G BASE-T/X  | SFP+ (10GE)  |
| 10G BASE-X  | SFP+ (10GE)  |
| 100/1000 BASE-T | 1000BASE-T (1GE)  |

If it's relevant, the manufacturer/model is Ericsson MINI-LINK 6691 / 6692 / 6693. HEre are some screenshots I have regarding the modules:

![image](https://github.com/netbox-community/netbox/assets/132988868/c302c4dc-1864-4df7-b04e-90b1f6b4c42e)

Thanks in advance

**Comment by mtinberg** [comment]
https://en.wikipedia.org/wiki/Ethernet_physical_layer
https://en.wikipedia.org/wiki/Small_Form-factor_Pluggable

I could see how those labels could be confusing

Something is ${max_speed}BASE-T if it is connected to Twisted-pair copper wiring (Cat3/5/6), T for Twisted.  Speeds on twisted-pair can go up to 10Gbit these days.  SFP (Small-formfactor-pluggable) is a 1Gbit (1000Mbit) standard and SFP+ a 10Gbit one that uses a small device that connects to a standard socket and outputs fibre optic (usually), twisted-pair or is built-in to a molded copper cable (using a fancier protocol than ethernet over twisted pair)

so you really have 5 distinct interface types and the labels are a bit off since the SFP/SFP+ isn't *BASE-T, but does have a max speed of 1G or 10G and not all 10G SFP+ ports are backwards compatible with 1G SFP modules (even on the same device), but most are.


so in your picture the two two on the top left are twisted-pair 1000Base-T and 2500Base-T, the two in the middle left TN5 and TN6 are 100Base-T and the two on the bottom left TN4 and TN5 are 1000Base-T

the rest are SFP and SFP+ depending on whether they support 1G or 10G, top two middle are SFP and SFP+, top two right are both SFP+, middle TN7, TN8, TN9 are SFP+ and bottom middle TN6 is SFP then TN7, TN8 are SFP+.

HTH

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: jayGG83 ***@***.***>
Sent: Wednesday, September 6, 2023 5:30 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Ethernet interface type equivalents in Netbox (Discussion #13696)


Hello,

I'd like to have your insight to check if my ethernet interface types mapping are correct. I'm trying to create microwave devices in Netbox (my knowledge is a bit limited as I'm only a developer, not a telco engineer) and some of the interface type names not really match with the interface type names in the devices and I'd like to verify if the types I selected are the correct ones. In case not, which would be the best option?

Interface type  Netbox equivalent
10/100 BASE-T   100BASE-T1
10/100/1000BASE-T       1000BASE-T (1GE)
100/1G BASE-T   1000BASE-T (1GE)
1000BASE-T/X    1000BASE-T (1GE)
1000/2500 BASE-T/X      2.5GBASE-T (2.5GE)
1G BASE-T/X     SFP
1G/10G BASE-T/X SFP+ (10GE)
10G BASE-X      SFP+ (10GE)
100/1000 BASE-T 1000BASE-T (1GE)

If it's relevant, the manufacturer/model is Ericsson MINI-LINK 6691 / 6692 / 6693. HEre are some screenshots I have regarding the modules:

[image]<https://user-images.githubusercontent.com/132988868/265978366-c302c4dc-1864-4df7-b04e-90b1f6b4c42e.png>

Thanks in advance

**Reply by jayGG83** [reply]
Thanks for the input.

I'm confused as to how to create these interfaces and setting the type in Netbox:

- For the 10/100 BASE-T, Netbox has: 100BASE-TX and 100 BASE-T1 as options inside Ethernet (fixed)
- For the 1000BASE-T/X, I used Netbox's ethernet interface type 1000BASE-T (1GE) option inside Ethernet (fixed), as I saw no other plausible match (I don't think the Ethernet (backplane) options might apply for any of these cases I have)
- For those that are SFP/SFP+, Netbox has two interface types for both; one inside Ethernet (modular) and another in FibreChannel

Thanks again

**Comment by mtinberg** [comment]
You probably want 100Base-TX for fixed 100Mb Ethernet but I wasn't familiar with 100Base-T1 so I looked it up, its an automotive version that works on a single copper twisted pair rather than distinct transmit and receive circuits.
https://en.wikipedia.org/wiki/Fast_Ethernet#100BASE-T1

1000Base-T (1G) is the right choice for gigabit Ethernet

You are configuring Ethernet for IP networking IIUC, not FibreChannel which is a different framing protocol for storage area networks (SANs) carrying something like SCSI.  You can use the same kind of modular optics though they just convert the bits into laser pulses, and you can track SAN switches in Netbox as well so FC options exist.

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
